src/librustdoc/html/format.rs

```diff
@@ -1582,7 +1586,10 @@ pub(crate) fn visibility_to_src_with_space<'a, 'tcx: 'a>(
             }
         }
     };
-    display_fn(move |f| f.write_str(&to_print))
+    display_fn(move |f| {
+        f.write_str(&hidden)?;
```

Comment:
Is there a need for storing this value here since you have a `bool`? What do you think about:

```suggestion
        if is_doc_hidden {
            f.write_str("#[doc(hidden)] ");
        }
```

Reply:
I wasn't sure how complex the closure was allowed to be. I'll make this change